File: 2020/Python/6.py
groups = []
group = []
with open("6.txt") as file:
    for line in file:
        if line != "\n":
            group.append(line.strip())
        else:
            groups.append(group)
            group = []
    groups.append(group)

counts = []
for group in groups:
    answered = []
    for questions in group:
        for letters in questions:
            if not letters in answered:
                answered.append(letters)
    counts.append(len(answered))

count = 0
for values in counts:
    count += values
print(count)

letters = "abcdefghijklmnopqrstuvwxyz"
questionCount = 0
for group in groups:
    for letter in letters:
        letterCount = 0
        for questions in group:
            if letter in questions:
                letterCount += 1
        if letterCount == len(group):
            questionCount += 1
print(questionCount)

File: 2020/Lua/6.lua
local q1, q2, count, count2, lcount = {}, {}, 0, 0, 0
for line in io.lines("6.txt") do
  if line == "" then
    for _ in pairs(q1) do
      count = count + 1
    end
    q1 = {}
    for _, total in pairs(q2) do
      if total == lcount then
        count2 = count2 + 1
      end
    end
    q2 = {}
    lcount = 0
  else
    lcount = lcount + 1
    for c in line:gmatch(".") do
      if q2[c] then q2[c] = q2[c] + 1 else q2[c] = 1 end
      q1[c] = true
    end
  end
end

for _ in pairs(q1) do
  count = count + 1
end

for _, total in pairs(q2) do
  if total == lcount then
    count2 = count2 + 1
  end
end
print(count)
print(count2)
